lending_contract/src/lib.rs

```diff
@@ -168,14 +163,52 @@ impl LendingNftCollateral {
     offers_vec.extend(reverse_vec.into_iter());
   }
 
+  fn cancel_specific_lending_offer(&mut self, offer_id: String, nft_collection_id: NftCollection) -> Option<bool> {
+    let nft_collection_lending_offers = self.lending_offers.get(&nft_collection_id);
+    let mut nft_collection_lending_offer_vec = self.lending_offers_vecs.get(&nft_collection_id).unwrap();
+    let specific_lending_offer = nft_collection_lending_offers.unwrap().get(&offer_id).unwrap();
+    if env::predecessor_account_id() == specific_lending_offer.owner_id {
+      // REORDER AND REMOVE FROM VECS
+      self.reorder_vec_without_specific_offer(&mut nft_collection_lending_offer_vec, specific_lending_offer.clone());
+      self.lending_offers.get(&nft_collection_id.clone()).unwrap().remove(&offer_id);
+      Some(true)
+    } else {
+      panic!("You are not the owner of this offer");
+    }
+  }
+
+  fn cancel_specific_borrowing_offer(&mut self, offer_id: String, nft_collection_id: NftCollection) -> Option<bool> {
```

Comment:
Mesma coisa em relação à anterior. Mas nesse caso eu sugiro retornar Promise. Quando retorna a promise, só vai indicar na wallet que a transação funcionou depois que a transferência do nft for concluída